### PR TITLE
#165413076 View notifications

### PR DESCRIPTION
**Pull request by AJAkimana**
Description
=======

This PR is for getting user notifications in the application

Type of change
=======
N/A

How has it been tested
=======
N/A

Checklist:
=======
- Design user interface
- Create action and reducer
- Implement getting user notifications
- Write tests

Pivotal tracker story ID
=======
[#165413076](https://www.pivotaltracker.com/n/projects/2326984/stories/165413076)

Screenshots
=======
Screen Shot 2019-07-24 at 10 51 05
Screen Shot 2019-07-25 at 17 35 47

### CODE REVIEW

**Review by Kabalisa — CHANGES_REQUESTED** [review]
@AJAkimana good work. but I am only able to see the number of notifications I have. not the notifications themselves. this is one part of what the story on pivotal tracker requires. can you please add the remaining functionality.

**Review by ericnyirimana — APPROVED** [review]

**Review by Kabalisa — APPROVED** [review]

File: src/actions/profile/getUserInfo.js [hunk 1]
```diff
@@ -31,3 +32,10 @@ export const getArticles = () => dispatch => {
     payload: http.get(`/api/v1/articles`)
   });
 };
+
+export const getNotifs = () => dispatch => {
+  dispatch({
+    type: GET_NOTIFS,
```
Comment (macphilips) [inline src/actions/profile/getUserInfo.js]: Use meaningful variable names

File: src/actions/profile/getUserInfo.js [hunk 2]
```diff
@@ -31,3 +32,10 @@ export const getArticles = () => dispatch => {
     payload: http.get(`/api/v1/articles`)
   });
 };
+
+export const getNotifs = () => dispatch => {
```
Comment (macphilips) [inline src/actions/profile/getUserInfo.js]: make it the name meaningful

File: src/components/common/userProfile.jsx [hunk 3]
```diff
@@ -1,53 +1,100 @@
...
+            <div className='list-notif'>
+              {!notifs.length ? (
+                <div className='notif-one'>
+                  <h4>No notifications</h4>
+                </div>
+              ) : (
+                  notifs.map((notif, index) => (
+                    <div className='notif-one' key={index}>
+                      <h4>
+                        {notif.notificationMessage} -{' '}
```
Comment (macphilips) [inline src/components/common/userProfile.jsx]:
```suggestion
                        {notif.notificationMessage}&nbsp;-&nbsp;
```

File: src/components/profile/viewProfile.jsx [hunk 4]
```diff
@@ -184,6 +184,17 @@ class ViewProfile extends Component {
                   />
                   <div className='bio'>{bio}</div>
                 </div>
+                {firstName || lastName ? (
+                  <span />
+                ) : (
+                  <div className='bio'>
+                    Update your profile: Click on the{' '}
```
Comment (macphilips) [inline src/components/profile/viewProfile.jsx]:
```suggestion
                    Update your profile: Click on the&nbsp;
```

File: src/components/profile/viewProfile.jsx [hunk 5]
```diff
@@ -184,6 +184,17 @@ class ViewProfile extends Component {
...
+                    <i>
+                      <b>Edit button</b>
+                    </i>{' '}
```
Comment (macphilips) [inline src/components/profile/viewProfile.jsx]:
```suggestion
                    </i>&nbsp;
```

File: src/tests/components/common/userProfile.test.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+      <UserProfile {...props} />
+    </MemoryRouter>
+  </Provider>
+);
+describe('<viewProfile/>', () => {
+  beforeEach(() => {
+    buttons = userProfile.find('img');
+
+    buttons.map(btn => btn.simulate('click', {}));
+    expect(userProfile).toHaveLength(1);
```
Comment (macphilips) [inline src/tests/components/common/userProfile.test.js:28]: why are expecting in a before each hook, this is not the standard practise please refactor this

File: src/tests/components/common/userProfile.test.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+);
+describe('<viewProfile/>', () => {
+  beforeEach(() => {
+    buttons = userProfile.find('img');
+    buttons.map(btn => btn.simulate('click', {}));
+  });
+  it('render components', () => {
+    expect(userProfile).toHaveLength(1);
+  });
+  it('componentWillReceiveProps', () => {
```
Comment (macphilips) [inline src/tests/components/common/userProfile.test.js:36]: What expected behavior are you testing here?

Comment (macphilips) [reply src/tests/components/common/userProfile.test.js:36]: @AJAkimana

Comment (AJAkimana) [reply src/tests/components/common/userProfile.test.js:36]: > What expected behavior are you testing here?

Let me fix it

**Pull request merged**